umbrella/package.yaml

```diff
@@ -0,0 +1,139 @@
+id: cisco-umbrella
+name: Cisco Umbrella
+author: Tenzir
+author_icon: https://raw.githubusercontent.com/tenzir/library/main/author.svg
+# package_icon: https://raw.githubusercontent.com/tenzir/library/main/umbrella/package.svg
+
+description: |
+  The [Cisco Umbrella](https://docs.umbrella.com/deployment-umbrella/docs/welcome-to-cisco-umbrella) package
+  onboards [DNS Log data](https://docs.umbrella.com/deployment-umbrella/docs/dns-log-formats) from
+  a [Cisco Umbrella S3 Bucket](https://docs.umbrella.com/deployment-umbrella/docs/log-management#logging-to-amazon-s3).
+
+inputs:
+  aws-access-key-id:
+    name: AWS Access Key ID
+    description: |
+      The AWS access key id.
+    # default: "XXXXXXXXXXXXXXXXXXXX"
+  aws-secret-access-key:
+    name: AWS Secret Access Key
+    description: |
+      The AWS secret access key.
+    # default: "YYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY"
+  aws-region:
+    name: AWS Region
+    description: |
+      The AWS Region of the log bucket.
+    default: eu-central-1
+  s3-url:
+    name: Prefix
+    description: |
+      The prefix (ie. subdirectory) under which to look for files in the S3 bucket.
+    # default: s3://cisco-managed-eu-central-1/0000000_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+  poll-frequency:
+    name: The poll frequency for the SIEM Integration API
+    description: |
+      How to often to check for new files
+    default: 15m
+
+pipelines:
+  onboard-events:
+    name: Onboard Cisco Umbrella DNS logs
+    description: |
+      The pipeline periodically checks the configured S3 bucket for new log files,
+      onboarding any that are detected. Any new files are downloaded, decompressed,
+      parsed, and published as `cisco.umbrella.dns` events onto the `cisco` topic.
+    definition: |
+      // tql2
+      every {{ inputs.poll-frequency }} {
+        shell r#"
+          # TODO: We want to switch to `remote {config}` as the first operator for
+          # more robust results.
+          export TENZIR_STATE_DIR=`tenzir --tql2 'config | select tenzir["state-directory"] | write_lines'`
+          export UV_CACHE_DIR=/tmp/cache/uv
+          cat >/tmp/script.py <<END
```

Comment:
For a future extension: This a unique path to avoid collisions when multiple instances or even versions run concurrently.